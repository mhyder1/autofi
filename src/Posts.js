import React, { useState, useEffect } from "react";
import "./Post.css";
import Comments from "./Comments";
export default function Posts() {
  const [expandedID, setExpandedID] = useState(1);
  const [posts, setPosts] = useState([]);

  const getPosts = async () => {
    const res = await fetch("https://jsonplaceholder.typicode.com/posts");
    const posts = await res.json();
    setPosts(posts);
  };

  useEffect(() => {
    getPosts();
  }, []);

  const displayPosts = () => {
    return posts.map(({ id, title, body }) => (
      <li className="posts" key={id}>
        <h2
          key={id}
          onClick={() => setExpandedID(id === expandedID ? null : id)}
        >
          {title} {/* posts */}
        </h2>
        <p>{body}</p>
        {expandedID === id && <Comments id={expandedID} />} {/* comments */}
      </li>
    ));
  };

  return <ul>{displayPosts()}</ul>;
}
