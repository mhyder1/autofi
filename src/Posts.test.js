import React from 'react';
import ReactDOM from 'react-dom';
import Posts from './Posts';
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme';

describe(`Posts Component`, () => {    
    it('Smoke test',() =>{
        const div = document.createElement('div');
        ReactDOM.render(<Posts/>,div);
        ReactDOM.unmountComponentAtNode(div);
    });
    
    it("Returns empty list item, if nothing passed.",() => {
        const wrapper = shallow(<Posts />)
        expect(toJson(wrapper)).toMatchSnapshot()
    });
})
