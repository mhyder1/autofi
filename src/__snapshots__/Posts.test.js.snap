// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts Component Returns empty list item, if nothing passed. 1`] = `<ul />`;
